### DISCUSSION TITLE
GetFileCopyBySourceFileNameAndAutoNumber

### DISCUSSION THREAD

**Discussion by DariuszMrozik** [koechlm/iLogic-Vault#18] [Q&A]
Hello,

Is there an option to copy model file with drawing like in Copy Design?
I.e.: GetFileCopyBySourceFileNameAndAutoNumber

Should I call method twice for IDW and model or there is another option?

Is there an method to check in checkouted file?
mVaultFile = iLogicVault.GetFileCopyBySourceFileNameAndAutoNumber(TemplateFile, mNumSchmName, mNumInput.ToArray , True)
 
Do something with mVaultFile

iLogicVaut.ChekInFIle(mVaultFile)

Should I use add file method even if it is a copy and already exist in Vault?

**Comment by koechlm** [comment]
Hello,
You asked multiple questions, and I try to answer these grouped by dependencies.

1. Is there an option... like in Copy Design: no - as of today, there is no option to include documentation files in the iLogicVault copy methods.
2. Should I call...: yes, so far, we resolved iLogic-Vault automation tasks in this way; create a copy for the model and the drawing. Replace the drawing's model with the copied and updated new model as an action of your rule within the running Inventor session.
3. Check-in of files checked out before: use the Inventor check-in command. iLogic-Vault provides no additional method to check in rather than sharing the rule code snippet calling the default command and a fully coded sample rule.
4. AddFile method targets supplemental file formats rather than Inventor files. For Inventor files, use the default available Vault commands.

Thank you for your questions. These shared some valuable feedback on missing information in the Intellisense descriptions; I will update these accordingly asap.
Best regards,
Markus

**Reply by DariuszMrozik** [reply]
Thank you for this information. I will try to use this methods.

**Comment by DariuszMrozik** [comment]
It would be good if there is additional option for checkin file/files by Vault path or local path. When I was trying to use inventor commands I need to ask user to accept it and also I need to open each document if I want to check in only some of them. Also additional method for copy with drawing will be great. Could you provide more info which method of Vault should I use to check file in (maybe I need to provide some references for checkin files, an if, then it will be complex).?

**Reply by koechlm** [reply]
I agree that the user commands need to have the appropriate Vault AddIn options: Set all dialogs to suppressed, and enable each workflow's option accordingly.
Check-in of Inventor files is complex, the file reference structure is one aspect, and the other one is writing the BOM information. If you follow the SDK documentation, you will see that this is not publicly documented; so I cannot integrate it into a publicly accessible source code. The best way to get there is to vote for the [Idea Station post](https://forums.autodesk.com/t5/vault-ideas/api-access-to-vault-inventor-add-in-and-other-cad-add-ins/idi-p/5703147) publishing an Inventor-Vault AddIn API.

**Comment by DariuszMrozik** [comment]
Ok, thank you. I will vote for an idea and ask more people to vote on it also.

Are you going to implement method to copy model with drawings?

**Reply by koechlm** [reply]
> Are you going to implement method to copy model with drawings?
No - I don't have planned to implement it. An iLogic rule is supposed to create copies for the model and drawing (applying one of the iLogic-Vault copy methods), then open and update all views with the new model.

**Reply by DariuszMrozik** [reply]
Ok, thank you.
Three more questions :)
1. Do you know how it's done copy drawings by copy design? Is is inventor needed to it?
2. How do you know if you need to update reference in drawing to more than one model? Sometimes user can add two parts on same drawing... Probably I will need to check what are existing references.
3. Do you have any code snippet for change reference and update view for iLogic?
